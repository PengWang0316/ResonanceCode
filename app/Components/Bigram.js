import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/Bigram.module.css';
import bigramBackground1 from '../imgs/bigramBackground1.jpg';
import bigramBackground2 from '../imgs/bigramBackground2.jpg';
import bigramBackground3 from '../imgs/bigramBackground3.jpg';
import bigramBackground4 from '../imgs/bigramBackground4.jpg';
import BigramCircle from './BigramCircle';
import BigramLine from './BigramLine';

const bigramBackground = {
  0: bigramBackground1,
  1: bigramBackground2,
  2: bigramBackground3,
  3: bigramBackground4
};
const bigramPosition = {
  0: ['leftMiddle', 'rightMiddle'],
  1: ['leftBottom', 'rightCenter'],
  2: ['left', 'right'],
  3: ['leftCenter', 'rightBottom']
};
const titlePosition = {
  0: [styles.titleTopLeft, styles.titleTopRight],
  1: [styles.titleBottomLeftA, styles.titleTopRightA],
  2: [styles.titleBottomLeft, styles.titleBottomRight],
  3: [styles.titleTopLeft, styles.titleBottomRight]
};
const bigramInfo = {
  0: [false, false, 'Taking Priority', 'Taking Priority'],
  1: [false, true, 'Giving Priority', 'Taking Priority'],
  2: [true, true, 'Giving Priority', 'Giving Priority'],
  3: [true, false, 'Taking Priority', 'Taking Priority']
};

/**
 * This component will use the information of line25 to decide where to show the arrows and lines' information.
 * Line's indicator 0: two solid lines  1: broken and solid   2: two brokens  3: solid and brokens
 *
 * The information of line46 will be used to show left circle.
 * The information of line13 will be used to show right circle.
*/

const Bigram = ({
  line25, line46, line13, isSimple
}) => (
  <div className="text-center">
    <img className={styles.bigramImg} src={bigramBackground[line25]} alt="Bigram background" />
    <BigramCircle circlePosition={bigramPosition[line25][0]} linePoint={line46} />
    <BigramCircle circlePosition={bigramPosition[line25][1]} linePoint={line13} isBrown />
    {!isSimple && <div className={`${titlePosition[line25][0]} d-flex align-items-center`}>Line 5:&nbsp;&nbsp; <BigramLine isBroken={bigramInfo[line25][0]} />&nbsp;&nbsp;{bigramInfo[line25][2]}</div>}
    {!isSimple && <div className={`${titlePosition[line25][1]} d-flex align-items-center`}>Line 2:&nbsp;&nbsp;<BigramLine isBroken={bigramInfo[line25][1]} />&nbsp;&nbsp;{bigramInfo[line25][3]}</div>}
  </div>
);
Bigram.propTypes = {
  line25: PropTypes.number.isRequired,
  line46: PropTypes.number.isRequired,
  line13: PropTypes.number.isRequired,
  isSimple: PropTypes.bool
};
Bigram.defaultProps = { isSimple: false };
export default Bigram;
