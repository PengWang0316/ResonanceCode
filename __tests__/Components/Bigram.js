import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import Bigram from '../../app/Components/Bigram';

describe('Bigram test', () => {
  const defaultProps = {
    line25: 0,
    line46: 1,
    line13: 2,
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<Bigram {...props} />);

  test('Bigram snapshot', () => expect(renderer.create(<Bigram {...defaultProps} />).toJSON()).toMatchSnapshot());
});
