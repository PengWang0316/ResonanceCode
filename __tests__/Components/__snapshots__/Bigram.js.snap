// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bigram test Bigram snapshot 1`] = `
<div
  className="text-center"
>
  <img
    alt="Bigram background"
    className="bigramImg"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="circle-leftMiddle"
  >
    <div
      className="internalDiv d-flex"
    >
      <div
        className="firstDiv"
      >
        <img
          alt="conception"
          className="ml-2 mt-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 0.4,
            }
          }
        />
      </div>
      <div
        className="secondDiv"
      >
        <img
          alt="seeding"
          className="mr-2 mt-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 0.4,
            }
          }
        />
      </div>
    </div>
    <div
      className="internalDiv d-flex"
    >
      <div
        className="thirdDiv"
      >
        <img
          alt="maturation"
          className="ml-2 mb-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 0.4,
            }
          }
        />
      </div>
      <div
        className="fourthDiv emphasizedOne"
      >
        <img
          alt="growth"
          className="mr-2 mb-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 1,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="circle-rightMiddle"
  >
    <div
      className="internalDiv d-flex"
    >
      <div
        className="firstDiv cancelBottomBorder"
      >
        <img
          alt="conception"
          className="ml-2 mt-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 0.4,
            }
          }
        />
      </div>
      <div
        className="secondDiv"
      >
        <img
          alt="seeding"
          className="mr-2 mt-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 0.4,
            }
          }
        />
      </div>
    </div>
    <div
      className="internalDiv d-flex"
    >
      <div
        className="thirdDiv emphasizedTwo"
      >
        <img
          alt="maturation"
          className="ml-2 mb-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 1,
            }
          }
        />
      </div>
      <div
        className="fourthDiv cancelLeftBorder"
      >
        <img
          alt="growth"
          className="mr-2 mb-2"
          src={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "opacity": 0.4,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="titleTopLeft d-flex align-items-center"
  >
    Line 5:   
    <div
      style={
        Object {
          "background": "#000",
          "height": "3px",
          "width": "30px",
        }
      }
    />
      Giving Priority
  </div>
  <div
    className="titleTopRight d-flex align-items-center"
  >
    Line 2:  
    <div
      className="d-flex"
      style={
        Object {
          "width": "30px",
        }
      }
    >
      <div
        style={
          Object {
            "background": "#000",
            "height": "3px",
            "width": "45%",
          }
        }
      />
      <div
        style={
          Object {
            "background": "#FFF",
            "height": "3px",
            "width": "10%",
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "height": "3px",
            "width": "45%",
          }
        }
      />
    </div>
      Giving Priority
  </div>
</div>
`;
